### PR TITLE
Port File Staging to VS17

### PR DESCRIPTION
**Pull request by matthchr**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by itowlson — APPROVED** [review]

File: src/SDKs/Batch/Support/FileStaging/Batch.FileStaging/FileStagingLinkedSources.cs [hunk 1]
```diff
@@ -13,20 +13,12 @@
...
-using System.Collections.Concurrent;
-using System.Diagnostics;
-using System.Threading;
-using System.Security;
-using System.Runtime.InteropServices;
 
 namespace Microsoft.Azure.Batch.FileStaging
 {
-    internal sealed class FileStagingLinkedSources
+    internal static class FileStagingLinkedSources
```
Comment (darylmsft) [inline src/SDKs/Batch/Support/FileStaging/Batch.FileStaging/FileStagingLinkedSources.cs]: probably no longer linked... the "link" referred to the vs feature.

**Review by darylmsft — APPROVED** [review]

File: src/SDKs/Batch/Support/FileStaging/Batch.FileStaging/FileStagingLinkedSources.cs [hunk 2]
```diff
@@ -44,20 +36,10 @@ private static string MakeDefaultNamePlusNamingFragment(string namingFragment)
...
-            lock (_lockForContainerNaming)
-            {
-                Thread.Sleep(30);  // make sure no two names are identical
-
-                string uniqueLetsHope = DateTime.UtcNow.ToString("yyyy-MM-dd-HH-mm-ss-fff");
-                string defContainerName = MakeDefaultNamePlusNamingFragment(namingFragment) + uniqueLetsHope;
-
-                return defContainerName;
-            }
+            string containerName = MakeDefaultNamePlusNamingFragment(namingFragment) + Guid.NewGuid().ToString("D");
```
Comment (matthchr) [inline src/SDKs/Batch/Support/FileStaging/Batch.FileStaging/FileStagingLinkedSources.cs]: I updated this to use a time (for ordering) _and_ a GUID for uniqueness at the end. Plus added some max length truncation where we will truncate the container name if it's too long.

That saves us the `Thread` (which we can't use) but I think maintains the original behavior.

**Comment by matthchr** [comment]
@shahabhijeet - This can be merged now (if CI passes)

**Comment by shahabhijeet** [comment]
@matthchr can you resolve the conflicts and push.

**Comment by matthchr** [comment]
@shahabhijeet Done

**Pull request merged**